### ISSUE TITLE
What happens if one of the shares is corrupt?

### ISSUE THREAD

**Issue by ekr** [abetterinternet/prio-documents#131]
We need some way for the helper to kick back an aggregate request with a bogus share without failing the entire batch.

**Comment by tgeoghegan** [comment]
IMO this is the same as #114: we need to ensure that the leader and helper(s) are aggregating over the same set of reports. The leader already gets an `AggregateSubResp` from helper for each report (or doesn't), so it should be able to intersect the set of valid reports leader has with the set of valid report helper has and then issue an `OutputShareReq` that only references those reports. Currently `OutputShareReq` only has `batch_start` and `batch_end`. I think it needs either a list of report IDs, or perhaps a list of report IDs to exclude if that is more efficient.

**Comment by cjpatton** [comment]
+1 to this being a duplicate.

@tgeoghegan's suggestion sounds a lot like what would be needed for the split-upload model that motivates #130. The main reason for the leader-only-upload model is that it avoids this cost by making the protocol simpler. The spec currently tries to solve this agreement problem by:
1. Having the aggregators run an input-validation protocol.
2. Having the leader hold the protocol state so that, at any given moment, it knows what set of reports are valid.
3. Ensuring that each report is aggregated at most once by enforcing a total ordering on reports.

We can speculate about failure scenarios and whether these measure are sufficient --- as we already have done, quite a lot! --- but from where I stand it's not clear yet that there's a problem to solve here. I suggest that we hold off on specifying anything new until we're confident that it's necessary.

**Comment by ekr** [comment]
Yes, now that I see there is an error, this seems like it should work OK.
